File: class/Formula.py
###Formule###
#-Accélération#
#-Vitesse#
#-Position#

class Formula:
    def acceleration(self, zeroToHundred, analogVertical):
        acceleration = (100/3.6)/zeroToHundred*analogVertical
        return acceleration
    
    def acceleration2(self, engineTorque, clutch, gearRatio, difRatio, wheelDiameter, mass, analogVertical):
        acceleration = engineTorque*clutch*gearRatio*difRatio/(wheelDiameter/2)/mass*analogVertical
        return acceleration
    
    def speed(self, acceleration, frameRate, speed):
        speed = acceleration*3.6/frameRate+speed
        return speed
        
    def position(self, acceleration, frameRate, speed, position):
        position = acceleration/2/frameRate+speed/frameRate+position
        return position

File: src/Vehicle/Formula.cpp
#include "Formula.hpp"

float Formula::acceleration(float engineTorque, float clutch, float gearRatio, float difRatio, float wheelDiameter, float mass, float analogVertical){
  //loger.log(engineTorque);
  //loger.log(wheelDiameter);
  float acceleration = engineTorque*clutch*gearRatio*difRatio/(wheelDiameter/2)/mass*analogVertical;
    return acceleration;
}

float Formula::speed(float acceleration, int frameRate, float Speed){
	float speed = acceleration*3.6/frameRate+Speed;
    return speed;
}

void Formula::vetor3Speed(Vector3& acceleration, int frameRate, Vector3& speed){

}

float Formula::position(float acceleration, int frameRate, float speed, float Position){
	float position = acceleration/2/frameRate+speed/frameRate+Position;
    return position;
}

void Formula::vector3Position(Vector3& acceleration, int frameRate, Vector3& speed, Vector3& position){

}
